inc/database_item.class.php

```diff
@@ -0,0 +1,425 @@
+      }
+      $canedit = $database->canEdit($ID);
+
+      $items = $DB->request([
+         'FROM'   => self::getTable(),
+         'WHERE'  => [
+            self::$items_id_1 => $ID
+         ]
+      ]);
+
+      Session::initNavigateListItems(
+         self::getType(),
+         //TRANS : %1$s is the itemtype name,
+         //        %2$s is the name of the item (used for headings of a list)
+         sprintf(
+            __('%1$s = %2$s'),
+            $database->getTypeName(1),
+            $database->getName()
+         )
+      );
+
+      if ($database->canAddItem('itemtype')) {
+         echo "<div class='firstbloc'>";
+         echo "<form method='post' name='databases_form$rand'
+                     id='databases_form$rand'
+                     action='" . Toolbox::getItemTypeFormURL(__CLASS__) . "'>";
+
+         echo "<table class='tab_cadre_fixe'>";
+         echo "<tr class='tab_bg_2'>";
+         echo "<th colspan='2'>" .
+               __('Add an item') . "</th></tr>";
+
+         echo "<tr class='tab_bg_1'><td class='center'>";
+         Dropdown::showSelectItemFromItemtypes(
+               ['items_id_name'   => 'items_id',
+                'itemtypes'       => Database::getTypes(true),
+                'entity_restrict' => ($database->fields['is_recursive']
+                                      ? getSonsOf('glpi_entities',
+                                       $database->fields['entities_id'])
+                                       : $database->fields['entities_id']),
+                'checkright'      => true,
+               ]);
+         echo "</td><td class='center' class='tab_bg_1'>";
+         echo Html::hidden('databases_id', ['value' => $ID]);
+         echo Html::submit(_x('button', 'Add'), ['name' => 'add']);
+         echo "</td></tr>";
+         echo "</table>";
+         Html::closeForm();
+         echo "</div>";
+      }
+
+      $items = iterator_to_array($items);
+
+      if (!count($items)) {
+         echo "<table class='tab_cadre_fixe'><tr><th>".__('No item found')."</th></tr>";
+         echo "</table>";
+      } else {
+         if ($canedit) {
+            Html::openMassiveActionsForm('mass'.__CLASS__.$rand);
+            $massiveactionparams = [
+               'num_displayed'   => min($_SESSION['glpilist_limit'], count($items)),
+               'container'       => 'mass'.__CLASS__.$rand
+            ];
+            Html::showMassiveActions($massiveactionparams);
+         }
+
+         echo "<table class='tab_cadre_fixehov'>";
+         $header = "<tr>";
+         if ($canedit) {
+            $header .= "<th width='10'>";
+            $header .= Html::getCheckAllAsCheckbox('mass'.__CLASS__.$rand);
+            $header .= "</th>";
+         }
+         $header .= "<th>".__('Itemtype')."</th>";
+         $header .= "<th>"._n('Item', 'Items', 1)."</th>";
+         $header .= "</tr>";
+         echo $header;
+
+         foreach ($items as $row) {
+            $item = new $row['itemtype'];
```

Comment:
Would be safer to check itemtype validity.
```suggestion
            if (!($item = getItemForItemtype($row['itemtype']))) {
               continue;
            }
```